src/rider/main/kotlin/me/fornever/avaloniarider/actions/StartAvaloniaPreviewerAction.kt

```diff
@@ -106,6 +122,43 @@ private fun startListeningTask(logger: Logger, serverSocket: ServerSocket) = Thr
     }
 }.apply { start() }
 
+fun handleMessage(logger: Logger, writer: BsonStreamWriter, message: AvaloniaMessage, targetPath: Path) {
+    when (message) {
+        is StartDesignerSessionMessage -> {
+            // hardcoded some xaml
+            val xaml = """
+            <Window xmlns="https://github.com/avaloniaui"
+                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+                    Title="AvaloniaTest">
+                Hello World!
+            </Window>
+            """.trimIndent()
+
+            val msg = UpdateXamlMessageBuilder.build(xaml, targetPath.toString())
+            writer.sendMessage(msg)
+        }
+        is UpdateXamlResultMessage -> message.error?.let {
+            logger.error { "Error from UpdateXamlResultMessage: $it" }
+        }
+        is RequestViewportResizeMessage -> {
+            window.size = Dimension(message.width.toInt(), message.height.toInt())
+
+            val dpi = 96.0
+            writer.sendMessage(ClientRenderInfoMessage(dpi, dpi))
+            writer.sendMessage(ClientViewportAllocatedMessage(message.width, message.height, 92.0, 92.0))
```

Comment:
WTF, is it 92 or 96?

Reply:
Seems that it should be 96 * 10, but I am not sure.
Saw it here https://github.com/AvaloniaUI/Avalonia/blob/022fa47397fc56cfbba01524afa724cac2929cc7/src/Avalonia.Controls/Remote/RemoteWidget.cs
and changed in commit https://github.com/ForNeVeR/AvaloniaRider/pull/7/commits/783a9d74ef1b78022f56b32add98b402d76b0237

Reply:
Yeah 96 is right, let's make it 96 here. 92 was my mistake.

Reply:
Ok, changed to 10 in commit https://github.com/ForNeVeR/AvaloniaRider/pull/7/commits/8297e0c7f768b3d1c328bd1a21992b4ee168d69f